masu/processor/_tasks/download.py

```diff
@@ -85,15 +87,22 @@ def _get_report_files(customer_name,
         disk_msg = 'Unable to find avaiable disk space. {} does not exist'.format(Config.TMP_DIR)
     LOG.info(disk_msg)
 
+    reports = None
     try:
         downloader = ReportDownloader(customer_name=customer_name,
                                       access_credential=authentication,
                                       report_source=billing_source,
                                       provider_type=provider_type,
                                       provider_id=provider_id,
                                       report_name=report_name)
-        return downloader.get_reports(number_of_months)
+        reports = downloader.get_reports(number_of_months)
     except (MasuProcessingError, MasuProviderError, ReportDownloaderError) as err:
```

Comment:
I think we need to revisit these exceptions and where they are thrown, under what conditions etc.  I tried many different things to get the this code to be hit and where your new logic would kick in.  I couldn't figure out a way.  I tried using an S3 bucket and AWS local to mess up the manifest so it fails and removing all contents in the directory.  Nothing worked.  I stayed in the `READY` state the whole time.

In fact it would be nice to have a class that could validate that a provider's backend configuration.  That would be a great status check for us but also could be used as the ingest checkpoint to decide whether or not the download should be attempted.

If we didn't do a configuration check prior to ingest; We need to make sure the backoff logic kicks in only for user configuration error situations where the user has misconfigured something.  We do a decent job of confirming some of these things when the provider is added in koku.  But once the provider exists we have no checks to confirm that those settings are still set.  Any other errors that is outside of the control of the user we should not be backing off on since it's an internal problem that we would need to fix.

Reply:
A simple way to force an error is to add a provider with a valid S3 bucket configuration, get past the initial config check, then delete the S3 bucket.

The Boto exception that's thrown will be re-raised as a ReportDownloaderError. That will hit that block.

On the OCP side, there's even fewer validation checks, which makes testing that path much easier. Just enter an incorrect report name into the provider then try to download.